### ISSUE TITLE
ssl.match_hostname() is deprecated

### ISSUE THREAD

**Issue by igor-mendix** [nabla-c0d3/sslyze#627]
`ssl.match_hostname()` function used here: https://github.com/nabla-c0d3/sslyze/blob/49380c1c2b1d1b521ffd4314fc52ea3927afd599/sslyze/plugins/certificate_info/_cert_chain_analyzer.py#L296
is [deprecated since Python 3.7](https://docs.python.org/3.11/library/ssl.html#ssl.match_hostname), and is already absent in 3.12.

**Comment by mig5** [comment]
Is there any easy workaround for this? This is a problem running sslyze 5.2.0 on the latest Debian stable (12 aka Bookworm).

**Comment by blshkv** [comment]
```
    from sslyze.plugins.certificate_info._cert_chain_analyzer import (
  File "/usr/lib/python3.12/site-packages/sslyze/plugins/certificate_info/_cert_chain_analyzer.py", line 3, in <module>
    from ssl import CertificateError, match_hostname
ImportError: cannot import name 'match_hostname' from 'ssl' (/usr/lib/python3.12/ssl.py)
```

sslyze fails to start with python3_12. Please fix

**Comment by blshkv** [comment]
a potential "quick" workaround is to use "https://pypi.org/project/backports.ssl_match_hostname/", but I'm not sure.
And it's like going backwards

**Comment by kchodkiewicz** [comment]
there was the same issue in other project and I'd say there is a good solution: https://github.com/aiortc/aioquic/issues/368

tl;dr
- [urllib3](https://github.com/urllib3/urllib3/blob/main/src/urllib3/util/ssl_match_hostname.py) has a port of match_hostname (simple to incorporate)
- [service-identity](https://github.com/pyca/service-identity/tree/main) different implementation but seems to be more *proper* solution (a bit more complex to incorporate)

**Comment by nabla-c0d3** [comment]
This might get solved by https://github.com/nabla-c0d3/sslyze/issues/638

**Comment by nabla-c0d3** [comment]
Fixed in [v6.0.0](https://github.com/nabla-c0d3/sslyze/releases/tag/6.0.0).

**Comment by uN4sh** [comment]
I don't understand how it can be "simpler to process" to identify a CN/Hostnam mismatch now that the `leaf_certificate_subject_matches_hostname` attribute has been removed from the JSON output ?
